draft-ietf-quic-transport.md

```diff
@@ -3018,11 +3018,23 @@ elements on the network path, an endpoint verifies the following when an ACK
 frame is received:
```

Comment:
I think that we need a little restructuring here.

> To reduce the risk of bad ECN markings affecting the operation of an endpoint, an endpoint verifies the counts it receives when it receives new acknowledgements:
> 
> * \[bullets]
>
> This validation is only performed if the ACK frame increases the largest received packet number.  Reordered acknowledgments could have lower counter values and might not be successfully validated as a result.
>
> These counts might be inflated if acknowledgments are never received for packets that were successfully delivered.  If validation succeeds, an endpoint MUST increase its expected counter values to those it receives.

Note that I almost didn't use MUST on that last clause, because there is no interop loss from doing that; it only means that validation is less likely to catch bad markings.

I also removed the "If the sender does not have state to determine if a packet is newly acknowledged..."  Such a packet - such as a packet that is declared lost - will never be considered to be newly acknowledged.

Would you like this as a PR?

Reply:
I think the restructuring is fine. On a high level I agree with your removal of the if you are missing state. However, considering some of the discussion on the quicdev slack I do wonder if something along those lines needs to be included to ensure that people are not interpreting this as requiring one to maintain state further into history than what one otherwise would.

Reply:
Thanks for that.

The pseudocode in the recovery draft drops packets from the "in flight" structure when they are declared lost and only treats packets as "newly acknowledged" if they are in that structure.  I think that makes it clear enough.